### PR TITLE
fix: sms

### PR DESCRIPTION
**Pull request by LucasdRossi**
## What changed? Why?

* Fixed the issues found in the [SMS Bug Bash](https://docs.google.com/document/d/18qcdgKEC_pzWvSlfxm_3Z5AWbym_ToHeHVuoEDnkd3I/edit#heading=h.ax8w0rsa4btj)
* Added a campaign name field and uses it to not send the same campaign message to users that already received it
* Added logic to send a bulk welcome message to users who didn't receive it

## PlanetScale deploy request

[#45](https://app.planetscale.com/stand-with-crypto/swc-web/deploy-requests/45)

## How has it been tested?

- [x] Locally
- [x] Vercel Preview Branch
- [ ] Unit test
- [ ] Functional test

### CODE REVIEW

**Review by olavoparno — APPROVED** [review]

File: src/inngest/functions/sms/bulkSMSCommunicationJourney.ts [hunk 1]
```diff
@@ -44,158 +41,98 @@ export const bulkSMSCommunicationJourney = inngest.createFunction(
...
-        async () => {
-          let messagesSent = 0
-          for (const batch of phoneNumberChunks) {
-            messagesSent += await enqueueMessages(batch, smsBody, persist)
-          }
+    await step.run('log-infos', () => ({
+      'Time to send all messages': formatTime(timeInSecondsToSendAllMessages),
+      'Total phone numbers': totalPhoneNumbers,
+      'Total batches': phoneNumberQueueBatches.length,
+    }))
```
Comment (lucasrmp-cb) [inline src/inngest/functions/sms/bulkSMSCommunicationJourney.ts:91]: We can use an inngest logger or configure ours in inngest to remove this workaround
https://www.inngest.com/docs/guides/logging#using-your-preferred-logger

File: src/inngest/functions/sms/bulkSMSCommunicationJourney.ts [hunk 2]
```diff
@@ -44,158 +41,98 @@ export const bulkSMSCommunicationJourney = inngest.createFunction(
...
-        hasNumbersLeft = false
-      }
+      iteration += 1
 
-      loopIteration += 1
+      await step.run('log-infos', () => ({
+        Messages: `${totalMessagesSent}/${totalPhoneNumbers}`,
+        Time: `${formatTime(totalTimeTaken)}/${formatTime(timeInSecondsToSendAllMessages)}`,
+        Iterations: `${iteration}/${phoneNumberQueueBatches.length}`,
+      }))
```
Comment (lucasrmp-cb) [inline src/inngest/functions/sms/bulkSMSCommunicationJourney.ts:129]: Same here

File: src/utils/server/sms/actions.ts [hunk 3]
```diff
@@ -3,20 +3,28 @@ import 'server-only'
 import { SMSStatus, User } from '@prisma/client'
 import { waitUntil } from '@vercel/functions'
 
-import { GOODBYE_SMS_COMMUNICATION_JOURNEY_INNGEST_EVENT_NAME } from '@/inngest/functions/sms/goodbyeSMSCommunicationJourney'
-import { UNSTOP_CONFIRMATION_SMS_COMMUNICATION_JOURNEY_INNGEST_EVENT_NAME } from '@/inngest/functions/sms/unstopConfirmationSMSCommunicationJourney'
+// import { GOODBYE_SMS_COMMUNICATION_JOURNEY_INNGEST_EVENT_NAME } from '@/inngest/functions/sms/goodbyeSMSCommunicationJourney'
+// import { UNSTOP_CONFIRMATION_SMS_COMMUNICATION_JOURNEY_INNGEST_EVENT_NAME } from '@/inngest/functions/sms/unstopConfirmationSMSCommunicationJourney'
```
Comment (lucasrmp-cb) [inline src/utils/server/sms/actions.ts:7]: Can we add that todo here too?

File: src/app/api/public/sms/events/messages/route.ts [hunk 4]
```diff
@@ -70,30 +71,32 @@ export async function POST(request: NextRequest) {
...
+      // We can't get the messageId when replying with twilio, so we need to trigger a Inngest function instead
       await optOutUser(phoneNumber, keyword === SWC_STOP_SMS_KEYWORD, user)
     } else if (['YES', 'START', 'CONTINUE', 'UNSTOP', SWC_UNSTOP_SMS_KEYWORD].includes(keyword)) {
       await optUserBackIn(phoneNumber, user)
     } else if (['HELP'].includes(keyword)) {
-      message = messages.HELP_MESSAGE
+      // We don't want to track this message, so we can just reply with twilio
+      // TODO: Uncomment this after we start using Messaging Service
+      // message = messages.HELP_MESSAGE
     }
```
[the remark this answers is not in the document]
Comment (LucasdRossi) [reply src/app/api/public/sms/events/messages/route.ts:90]: We still need to call `optOutUser` and `optUserBackIn` to change their smsStatus field

File: src/inngest/functions/sms/bulkSMSCommunicationJourney.ts (hunk 1 above)
Comment (LucasdRossi) [reply src/inngest/functions/sms/bulkSMSCommunicationJourney.ts:91]: As this is not related to this task and involves changing all other inngest functions I created a ticket to work on this [#125](https://github.com/Stand-With-Crypto/swc-internal/issues/125)

**Review by lucasrmp-cb — DISMISSED** [review]

**Review by lucasrmp-cb — APPROVED** [review]

**Review by olavoparno — APPROVED** [review]

**Pull request merged**